### PR TITLE
Ammunition calibers PR and the crash that cause a pain in my ass

### PR DESCRIPTION
**Pull request by PatoGrone**
Ammunition calibers PR:
added 11 different calibers with 12 magazines and 11 ammo boxes for all of them.
added 2 guns, the riot shotgun, just a combat shotgun chambered on 20 gauge, and the pipe rifle, chambered in .308. in a future PR i will add the ammo variants (AP, HP, FMJ, etc).
i gotta inform that i have a problem at the moment of insert any cartride or magazine inside of a gun, causing to crash my tests plays. i cant really see what is causing that and i need extra eyes has always
Captura

### CODE REVIEW

File: Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/50.yml
Comment (Peptide90) [inline Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/50.yml:1]: What da file doin

File: Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Ammunition/Magazines/10mm.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  abstract: true
+  components:
+  - type: Tag
+    tags:
+      - MagazinePistol
+  - type: BallisticAmmoProvider
+    mayTransfer: true
+    whitelist:
+      tags:
+        - N14CartridgePistol10
```
Comment (Peptide90) [inline Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Ammunition/Magazines/10mm.yml:56]: Need to make sure there’s a tag entry for this.

**Review by Peptide90 — CHANGES_REQUESTED** [review]
Quick scan, found two things, the latter is probably your crash. 

You need to create tag prototypes for any new tags added. These are probably in resources/prototypes/Nuclear14/tags.yml

File: Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Rifles/rifles.yml [hunk 2]
```diff
@@ -158,20 +159,66 @@
...
+  - type: Appearance
+
+- type: entity
+  name: Service Rifle
+  parent: BaseWeaponRifle
+  id: N14WeaponService
+  description: A cheap mass produced post war rifle made by the NCR, chambered in 5.56.
+  components:
+  - type: Sprite
+    sprite: Nuclear14/Objects/Weapons/Guns/Rifles/servicerifle.rsi
```
Comment (Peptide90) [inline Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Rifles/rifles.yml:189]: Sprite isn’t in this PR

File: Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Rifles/rifles.yml [hunk 3]
```diff
@@ -158,20 +159,66 @@
...
+    layers:
+      - state: base
+        map: ["enum.GunVisualLayers.Base"]
+      - state: mag-0
+        map: ["enum.GunVisualLayers.Mag"]
+  - type: Item
+    sprite: Nuclear14/Objects/Weapons/Guns/Rifles/servicerifle.rsi
```
Comment (Peptide90) [inline Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Rifles/rifles.yml:196]: Same here

**Review by Peptide90 — CHANGES_REQUESTED** [review]

**Review by Cheackraze — APPROVED** [review]

**Pull request merged**